### PR TITLE
Adding coverage and publish docs to gitlab-ci

### PR DESCRIPTION
**Pull request by niamhdougan**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
+  coverage: '/TOTAL.*\s+(\d+\%)/'
   before_script:
     - pipenv install --dev
   script:
     - pipenv run tests
     - pipenv run docs
     - dls-python3 setup.py bdist_wheel
     - dls-py3 check --no-tag-check
+  after_script:
+    - pipenv run coverage html
```
Comment (thomascobb) [inline .gitlab-ci.yml:17]: Not convinced we need this line, doesn't the coverage output come out of pipenv run tests?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -1,10 +1,20 @@
 rhel7_job:
   tags:
     - rhel7
+  variables:
+    DOCS_BUILD_DIR: build/html
+    DOCS_PUBLISH_ROOT: /dls/cs-www/reports/gitlab-ci/${CI_PROJECT_NAME}/docs
+    DOCS_PUBLISH_URL: http://www.cs.diamond.ac.uk/reports/gitlab-ci/${CI_PROJECT_NAME}/docs
```
Comment (thomascobb) [inline .gitlab-ci.yml:7]: This variable doesn't seem to be used

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -1,10 +1,20 @@
 rhel7_job:
   tags:
     - rhel7
+  variables:
+    DOCS_BUILD_DIR: build/html
```
Comment (thomascobb) [inline .gitlab-ci.yml:5]: I'd probably delete this variable and move its value into the cp command as it's only used once

File: Pipfile [hunk 4]
```diff
@@ -17,7 +17,7 @@ doc8 = "*"
 [scripts]
 # Put coverage here so we don't interfere with debugging in the IDE
 tests = "python -m pytest --cov-fail-under=90 --cov=dls_python3_template_module --cov-report term"
-docs = "sphinx-build -E -b html docs build/html"
+docs = "sphinx-build -b html docs build/html"
```
Comment (thomascobb) [inline Pipfile:20]: Did you mean to delete this option?

File: .gitlab-ci.yml (hunk 1 above)
Comment (niamhdougan) [reply .gitlab-ci.yml:17]: Yep, this was leftover from when I tried to publish coverage. Will remove.

File: Pipfile (hunk 4 above)
Comment (niamhdougan) [reply Pipfile:20]: I think so, it just builds the docs from scratch, worth keeping?

Comment (thomascobb) [reply Pipfile:20]: Probably, I normally put it on otherwise it doesn't rebuild API docs when docstrings change...

**Pull request merged**